File: frontend/src/types/monitorIndex.ts
import type { ApiResponse, PageResult } from '@/types/department'

export interface MonitorIndex {
  id?: number
  departmentId: number
  type: number        // 1=环境, 2=作用, 3=结构响应, 4=结构变化
  name: string
  code?: string       // 编码
  createdAt?: string
  updatedAt?: string
}

export interface MonitorValueType {
  id?: number
  monitorIndexId: number
  name: string
  unit?: string
  createdAt?: string
  updatedAt?: string
}

export type { ApiResponse, PageResult }

File: backend/src/main/java/com/easyshm/entity/MonitorIndex.java
package com.easyshm.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "monitor_index")
public class MonitorIndex {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "department_id", nullable = false)
    private Long departmentId;

    /** 1=环境, 2=作用, 3=结构响应, 4=结构变化 */
    @Column(nullable = false)
    private Integer type;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 50)
    private String code;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
